src/core/reader/types/values.rs

```diff
@@ -14,14 +14,19 @@ use crate::{Error, Result};
 impl WasmReader<'_> {
     /// Note: If `Err`, the [WasmReader] object is no longer guaranteed to be in a valid state
     pub fn read_u8(&mut self) -> Result<u8> {
-        let value = *self.current.first().ok_or(Error::Eof)?;
+        if self.full_contents.len() <= self.pc {
```

Comment:
You can use the peek_u8() to get the value and check if the pc is valid at the same time